src/rust/engine/options/src/env_tests.rs

```diff
@@ -15,6 +16,39 @@ fn env<I: IntoIterator<Item = (&'static str, &'static str)>>(vars: I) -> Env {
   }
 }
 
+#[test]
+#[cfg(not(target_os = "windows"))]
+fn test_capture_lossy() {
```

Comment:
Ok. This sort of test is a bit tautological, I was thinking an IT to actually show you could run pants in an env with bad keys and values, but it doesn't hurt too much.

Reply:
I don't think it's tautological - it tests that we correctly classify env vars, and don't crash. True though that it does assume that this is the production pathway for reading env vars.

Still, I also added an IT.

Reply:
(and verified that it fails in the expected way on main)